sdk/python/kfp/compiler/_component_builder.py

```diff
@@ -104,10 +109,10 @@ def __init__(self):
     self._dependency = {self._PYTHON_PACKAGE:OrderedDict()}
 
   @property
-  def python_package(self):
+  def python_packages(self):
     return self._dependency[self._PYTHON_PACKAGE]
 
-  def add_python_package(self, name, version=DependencyVersion(), override=True):
+  def add_python_package(self, version=VersionedDependency(name='default'), override=True):
```

Comment:
1. The `default` dependency does not seem to make sense. It should be a required parameter. If your intent is to put a hint about a type, then use a type hint (`dependency: VersionedDependency`)
2. `version` --> `dependency` (`version.name` looks really strange)

Reply:
make sense.

Reply:
Done